testing/internal/e2e/tests/base/paginate_credential_test.go

```diff
@@ -0,0 +1,216 @@
+		)
+	}
+
+	// List credentials
+	output := e2e.RunCommand(ctx, "boundary",
+		e2e.WithArgs(
+			"credentials", "list",
+			"-credential-store-id", newStoreId,
+			"-format=json",
+		),
+	)
+	require.NoError(t, output.Err, string(output.Stderr))
+
+	var initialCredentials credentials.CredentialListResult
+	err = json.Unmarshal(output.Stdout, &initialCredentials)
+	require.NoError(t, err)
+
+	var returnedIds []string
+	for _, cred := range initialCredentials.Items {
+		returnedIds = append(returnedIds, cred.Id)
+	}
+
+	require.Len(t, initialCredentials.Items, c.MaxPageSize+1)
+	assert.Empty(t, cmp.Diff(returnedIds, credentialIds, cmpopts.SortSlices(func(i, j string) bool { return i < j })))
+	assert.Empty(t, initialCredentials.ResponseType)
+	assert.Empty(t, initialCredentials.RemovedIds)
+	assert.Empty(t, initialCredentials.ListToken)
+
+	// Create a new credential and destroy one of the other credentials
+	newCredentialId := boundary.CreateNewStaticCredentialPasswordCli(t, ctx, newStoreId, "user", "password")
+	output = e2e.RunCommand(ctx, "boundary",
+		e2e.WithArgs(
+			"credentials", "delete",
+			"-id", initialCredentials.Items[0].Id,
+		),
+	)
+	require.NoError(t, output.Err, string(output.Stderr))
+
+	// List again, should have the new credential but not the deleted credential
+	output = e2e.RunCommand(ctx, "boundary",
+		e2e.WithArgs(
+			"credentials", "list",
+			"-credential-store-id", newStoreId,
+			"-format=json",
+		),
+	)
+	require.NoError(t, output.Err, string(output.Stderr))
+
+	var newCredentials credentials.CredentialListResult
+	err = json.Unmarshal(output.Stdout, &newCredentials)
+	require.NoError(t, err)
+
+	require.Len(t, newCredentials.Items, c.MaxPageSize+1)
+	// The first item should be the most recently created, which
+	// should be our new credential
+	firstItem := newCredentials.Items[0]
+	assert.Equal(t, newCredentialId, firstItem.Id)
+	assert.Empty(t, newCredentials.ResponseType)
+	assert.Empty(t, newCredentials.RemovedIds)
+	assert.Empty(t, newCredentials.ListToken)
+	// Ensure the deleted credential isn't returned
+	for _, credential := range newCredentials.Items {
+		assert.NotEqual(t, credential.Id, initialCredentials.Items[0].Id)
+	}
+}
+
+// TestApiPaginateCredentials asserts that the API automatically paginates to retrieve
+// all Credentials in a single invocation.
+func TestApiPaginateCredentials(t *testing.T) {
+	e2e.MaybeSkipTest(t)
+	c, err := loadTestConfig()
+	require.NoError(t, err)
+
+	client, err := boundary.NewApiClient()
+	require.NoError(t, err)
+	ctx := context.Background()
+	sClient := scopes.NewClient(client)
+	cClient := credentials.NewClient(client)
+	newOrgId := boundary.CreateNewOrgApi(t, ctx, client)
+	t.Cleanup(func() {
+		ctx := context.Background()
+		_, err := sClient.Delete(ctx, newOrgId)
+		require.NoError(t, err)
+	})
+	newProjectId := boundary.CreateNewProjectApi(t, ctx, client, newOrgId)
+	newStoreId := boundary.CreateNewCredentialStoreStaticCli(t, ctx, newProjectId)
```

Comment:
I think we want to use the method `CreateNewCredentialStoreStaticApi` instead?

Reply:
Sure, can do!